clang/lib/CodeGen/CodeGenModule.cpp

```diff
@@ -4098,8 +4098,26 @@ void CodeGenModule::emitMultiVersionFunctions() {
     }
 
     llvm::Constant *ResolverConstant = GetOrCreateMultiVersionResolver(GD);
-    if (auto *IFunc = dyn_cast<llvm::GlobalIFunc>(ResolverConstant))
+    if (auto *IFunc = dyn_cast<llvm::GlobalIFunc>(ResolverConstant)) {
       ResolverConstant = IFunc->getResolver();
+      // In Aarch64, default versions of multiversioned functions are mangled to
+      // their 'normal' assembly name. This deviates from other targets which
+      // append a '.default' string. As a result we need to continue appending
+      // .ifunc in Aarch64.
+      // FIXME: Should Aarch64 mangling for 'default' multiversion function and
+      // in turn ifunc function match that of other targets?
+      if (FD->isTargetClonesMultiVersion() &&
+          !getTarget().getTriple().isAArch64()) {
+        const CGFunctionInfo &FI = getTypes().arrangeGlobalDeclaration(GD);
+        llvm::FunctionType *DeclTy = getTypes().GetFunctionType(FI);
+        std::string MangledName = getMangledNameImpl(
+            *this, GD, FD, /*OmitMultiVersionMangling=*/true);
+        auto *Alias = llvm::GlobalAlias::create(
+            DeclTy, 0, llvm::Function::WeakODRLinkage, MangledName + ".ifunc",
```

Comment:
Why WeakODR?

Reply:
I based this PR off of https://reviews.llvm.org/D67058 where a similar alias was added for cpu_dispatch with WeakODRLinkage. I assumed it was to allow merging across TUs. The review has some discussion about this.

Reply:
I looked into this a bit and I think `WeakODR `is incorrect here. I see we call `getMultiversionLinkage` to set linkage of resolver function. I believe this is to account for internal linkage as well. I think that is the right solution here. I am not too familiar with Linkage Types though and so I could be wrong and any input is appreciated.

Reply:
That sounds right... we don't want to make symbols that are supposed to be internal linkage visible through an alias.

Reply:
Thank you for your feedback. I have made the change